hls4ml/csim_testbench/myproject_test.cpp

```diff
@@ -0,0 +1,299 @@
+//hls-fpga-machine-learning insert bram
+
+#define CHECKPOINT 5
+
+namespace nnet {
+    bool trace_enabled = true;
+    std::map<std::string, void *> *trace_outputs = NULL;
+    size_t trace_type_size = sizeof(double);
+}
+
+int main(int argc, char **argv)
+{
+  //load input data from text file
+  std::ifstream fin("tb_data/tb_input_features.dat");
+  //load ground truth class labels from text file
+  std::ifstream fpr("tb_data/tb_output_predictions.dat");
+  //load hls predictions from text file
+  std::ifstream fpr_hls("tb_data/tb_hls_predictions.dat");
+  //load keras predictions from text file
+  std::ifstream fpr_keras("tb_data/tb_keras_predictions.dat");
+
+#ifdef RTL_SIM
+  std::string RESULTS_LOG = "tb_data/rtl_cosim_results.log";
+#else
+  std::string RESULTS_LOG = "tb_data/csim_results.log";
+#endif
+  std::ofstream fout(RESULTS_LOG);
+
+  std::string iline;
+  std::string pline;
+  std::string pline_hls;
+  std::string pline_keras;
+  int e = 0;
+
+  //Variables to keep track of statistics
+  bool class_match = false;
+  int total_inputs  = 0;
+  int total_matches = 0;
+  std::vector<float> zeros (4, 0.0);
+  std::vector<std::vector<float>> total_m (4, zeros);
+  std::vector<std::vector<float>> curr_m (4, zeros);
+  std::vector<std::string> ids = {"gt", "hp", "k ", "hc"};
+
+  if (fin.is_open() && fpr.is_open() && fpr_hls.is_open() && fpr_keras.is_open()) {
+    while ( std::getline(fin,iline) && std::getline(fpr,pline) && std::getline(fpr_hls,pline_hls) && std::getline(fpr_keras,pline_keras)) {
+      if (e % CHECKPOINT == 0) {
+          std::cout << "Processing input " << e << ":"<< std::endl;
+      } 
+      //Loading input
+      char* cstr=const_cast<char*>(iline.c_str());
+      char* current;
+      std::vector<float> in;
+      current=strtok(cstr," ");
+      while(current!=NULL) {
+        in.push_back(atof(current));
+        current=strtok(NULL," ");
+      }
+
+      //Load ground truth class label into vector
+      cstr=const_cast<char*>(pline.c_str());
+      std::vector<float> pr;
+      current=strtok(cstr," ");
+      while(current!=NULL) {
+        pr.push_back(atof(current));
+        current=strtok(NULL," ");
+      }
+
+      //Load hls prediction into vector
+      cstr=const_cast<char*>(pline_hls.c_str());
+      std::vector<float> pr_hls;
+      current=strtok(cstr," ");
+      while(current!=NULL) {
+        pr_hls.push_back(atof(current));
+        current=strtok(NULL," ");
+      }
+
+      //Load keras prediction into vector
+      cstr=const_cast<char*>(pline_keras.c_str());
+      std::vector<float> pr_keras;
+      current=strtok(cstr," ");
+      while(current!=NULL) {
+        pr_keras.push_back(atof(current));
+        current=strtok(NULL," ");
+      }
+
+      //hls-fpga-machine-learning insert data
+      hls::stream<input_t> input_1("input_1");
+      nnet::copy_data<float, input_t, 0, N_INPUT_1_1*N_INPUT_2_1*N_INPUT_3_1>(in, input_1);
+      hls::stream<layer18_t> layer18_out("layer18_out");
```

Comment:
i if we change the model architecture the final layer would no longer be `layer18`... it would be nice if there was a way to make this less hard-coded...